feature/KAS 4526 nieuw dossier overzicht

kobemertens opened the pull request:
- remove cases loading page
- improve cases header
- add cases status pill
- update dossier page
- add submitters component
- add submitters filter
- use formatlist for submitters
- fix title filter

https://kanselarij.atlassian.net/browse/KAS-4526

kobemertens committed:
remove cases loading page

kobemertens committed:
improve cases header

kobemertens committed:
add cases status pill

kobemertens committed:
update dossier page

kobemertens committed:
add submitters component

kobemertens committed:
add submitters filter

kobemertens committed:
use formatlist for submitters

kobemertens committed:
fix title filter

kobemertens committed:
put dash in cells that miss a value

sergiofenoll committed:
Merge branch 'development' into feature/KAS-4526-nieuw-dossier-overzicht

sergiofenoll commented on app/components/cases/submitters.hbs:
```diff
@@ -0,0 +1,11 @@
+      {{formattedPart.value}}
+    </FormatList>
+  {{else}}
+    <p class="au-u-text-center">-</p>
```
The dashes look weird. They're correctly centered, it's just that the columns are a lot wider than they need to be, so the dashes don't end up in the middle of the text. I'd remove the center-alignment or try to figure out why the columns are so wide.

![image](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/assets/22411874/0894e069-8b7b-4e18-a430-5c0dfbde760a)

sergiofenoll replied:
Removed the centering. We can always bring it back later.
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/2045/commits/cb5b71ae046c402a5a7b1eca74c99f08e042c8b1

sergiofenoll committed:
chore: remove searchNameTextBuffer

By using the native HTML attribute value instead of the two-way binding
argument @value of AuInput, we don't need the buffer (the two-way
binding is what was causing the search to fire off on every key stroke,
because the variable was immediately being updated).

sergiofenoll committed:
chore: use app-wide debounce time for debounce of search in cases

sergiofenoll commented on app/components/cases/cases-header.js:
```diff
@@ -28,4 +29,17 @@ export default class CasesHeader extends Component {
+      this.clearFilter();
+      return;
+    }
+    await timeout(350);
```
There's an app-wide debounce time in `config/config.js` called `LIVE_SEARCH_DEBOUNCE_TIME`.

sergiofenoll replied:
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/2045/commits/56f833b0a303ea63b78e9f235f5adffba9b6bf3e

sergiofenoll commented on app/components/cases/cases-header.hbs:
```diff
@@ -1,36 +1,57 @@
+        <AuInput
+          @width="block"
+          @icon="search"
+          @value={{@nameSearchText}}
```
`@value` is going to do a two-way binding of the value, so that `@nameSearchText` gets immediately updated as soon as the input changes, which bypasses the `{{on "input" ... }}`.

We don't notice this because we don't actually use the buffer for anything, what actually causes the search to fire is the `setNameSearchText` method we pass in to this component. We can remove the buffer altogether by removing this two-way binding and using `value={{@nameSearchText}}`. This way the initial value is set, but it doesn't auto-update.

Removing the buffer is also beneficial because otherwise we need to add a `setupController` hook to set the `searchTextNameBuffer` field (otherwise when you e.g. reload the page when a filter is set, the input field won't be pre-filled with the right value because the buffer would be empty). By directly using the query param we don't need to take all of that into account.

sergiofenoll replied:
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/2045/commits/2924ea0abf3285b13c9c406d3899154e1d094482

sergiofenoll committed:
chore: remove text centering on dashes for empty values
